qualtran/bloqs/chemistry/thc/prepare.py

```diff
@@ -361,124 +368,138 @@ def selection_registers(self) -> Tuple[Register, ...]:
-        keep, sigma, less_than = bb.add(lte_gate, x=keep, y=sigma, target=less_than)
-        # delete the QROM
         # Select expects three plus states so set them up here.
-        plus_a = bb.add(Hadamard(), q=plus_a)
-        plus_b = bb.add(Hadamard(), q=plus_b)
-        plus_mn = bb.add(Hadamard(), q=plus_mn)
-        (nu_eq_mp1, plus_a), extra_ctrl = bb.add(
-            MultiControlX(cvs=(0, 1)), controls=np.array([nu_eq_mp1, plus_a]), target=extra_ctrl
+        soqs['plus_a'] = bb.add(Hadamard(), q=soqs['plus_a'])
+        soqs['plus_b'] = bb.add(Hadamard(), q=soqs['plus_b'])
+        soqs['plus_mn'] = bb.add(Hadamard(), q=soqs['plus_mn'])
+        (soqs['nu_eq_mp1'], soqs['plus_a']), soqs['extra_ctrl'] = bb.add(
+            MultiControlX(cvs=(0, 1)),
+            controls=np.array([soqs['nu_eq_mp1'], soqs['plus_a']]),
+            target=soqs['extra_ctrl'],
         )
-        extra_ctrl, mu, nu = bb.add(CSwap(bitsize=log_mu), ctrl=extra_ctrl, x=mu, y=nu)
-        out_regs = {
-            'mu': mu,
-            'nu': nu,
-            'plus_mn': plus_mn,
-            'plus_a': plus_a,
-            'plus_b': plus_b,
-            'sigma': sigma,
-            'rot': rot,
-            'succ': succ,
-            'nu_eq_mp1': nu_eq_mp1,
-            'theta': theta,
-            's': s,
-            'alt_mn': [alt_mu, alt_nu],
-            'alt_theta': alt_theta,
-            'keep': keep,
-            'less_than': less_than,
-            'extra_ctrl': extra_ctrl,
-        }
-        return out_regs
+        soqs['extra_ctrl'], soqs['mu'], soqs['nu'] = bb.add(
+            CSwap(bitsize=log_mu), ctrl=soqs['extra_ctrl'], x=soqs['mu'], y=soqs['nu']
+        )
+        return soqs
 
     def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
         cost_1 = (UniformSuperpositionTHC(self.num_mu, self.num_spin_orb), 1)
         nmu = self.num_mu.bit_length()
         data_size = self.num_spin_orb // 2 + self.num_mu * (self.num_mu + 1) // 2
         nd = (data_size - 1).bit_length()
         cost_2 = (ToContiguousIndex(nmu, nd), 1)
-        qroam = SelectSwapQROM.build_from_data(
-            *(self.theta, self.alt_theta, self.alt_mu, self.alt_nu, self.keep),
-            target_bitsizes=(1, 1, nmu, nmu, self.keep_bitsize),
-            use_dirty_ancilla=False,
-        )
+        qroam = self.build_qrom_bloq()
         cost_3 = (qroam, 1)
         cost_4 = (OnEach(self.keep_bitsize, Hadamard()), 1)
         cost_5 = (LessThanEqual(self.keep_bitsize, self.keep_bitsize), 2)
         cost_6 = (CSwap(nmu), 3)
-        cost_7 = (Toffoli(), 1)
-        return {cost_1, cost_2, cost_3, cost_4, cost_5, cost_6, cost_7}
+        cost_7 = (MultiControlX(cvs=(0, 1)), 1)
+        cost_8 = (XGate(), 2)
+        cost_9 = (CZ(), 2)
+        cost_10 = (Hadamard(), 3)
+        return {cost_1, cost_2, cost_3, cost_4, cost_5, cost_6, cost_7, cost_8, cost_9, cost_10}
```

Comment:
nit: Please use a dictionary instead of set, since set representation will be deprecated soon.

Reply:
Done.